tensorflow/contrib/go/example_test.go

```diff
@@ -0,0 +1,192 @@
+package tensorflow_test
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+func ExampleGraph_Op() {
+	var out []*tensorflow.Tensor
+
+	additions := 10
+	inputSlice1 := []int32{1, 2, 3, 4}
+	inputSlice2 := []int32{5, 6, 7, 8}
+
+	graph := tensorflow.NewGraph()
+	input1, _ := graph.Variable("input1", inputSlice1)
+	input2, _ := graph.Constant("input2", inputSlice2)
+
+	add, _ := graph.Op("Add", "add_tensors", []*tensorflow.GraphNode{input1, input2}, "", map[string]interface{}{})
+	graph.Op("Assign", "assign_inp1", []*tensorflow.GraphNode{input1, add}, "", map[string]interface{}{})
+
+	s, _ := tensorflow.NewSession()
+	s.ExtendAndInitializeAllVariables(graph)
+
+	for i := 0; i < additions; i++ {
+		out, _ = s.Run(nil, []string{"input1"}, []string{"assign_inp1"})
+	}
+
+	for i := 0; i < len(inputSlice1); i++ {
+		val, _ := out[0].GetVal(i)
+		fmt.Println("The result of: %d + (%d*%d) is: %d", inputSlice1[i], inputSlice2[i], additions, val)
+	}
+}
+
+func ExampleNewTensor_slice() {
+	tensorflow.NewTensor([][]int64{
+		{1, 2, 3, 4},
+		{5, 6, 7, 8},
+	})
+}
+
+func ExampleNewTensor_scalar() {
+	tensorflow.NewTensor("Hello TensorFlow")
+}
+
+func ExampleNewGraphFromText() {
+	graph, err := tensorflow.NewGraphFromReader(strings.NewReader(`
+		node {
+			name: "output"
+			op: "Const"
+			attr {
+				key: "dtype"
+				value {
+					type: DT_FLOAT
+				}
+			}
+			attr {
+				key: "value"
+				value {
+					tensor {
+						dtype: DT_FLOAT
+						tensor_shape {
+						}
+						float_val: 1.5 
+					}
+				}
+			}
+		}
+		version: 5`), true)
+
+	fmt.Println(graph, err)
+}
+
+func ExampleGraph_Constant() {
+	graph := tensorflow.NewGraph()
+	// Add scalar string to the Graph named 'const1'.
```

Comment:
// Add a scalar string node named 'const1' to the Graph.
(otherwise, it sounds like the Graph is named const1. :)